pkg/agent/toggles/types.go

```diff
@@ -72,19 +77,32 @@ func (t *Toggles) getMap(name string, entity *Entity) map[string]string {
 		log.Printf("map toggles are nil, resolving to default empty map value for toggle: %q", name)
 		return map[string]string{}
 	}
+	value := map[string]string{}
 	if toggle, ok := t.Maps[name]; ok {
-		return toggle(entity)
+		value = toggle(entity)
 	}
-	return map[string]string{}
+	log.Printf("resolved value of toggle %q; entity: %s; value: %s", name, entity, marshalToString(value))
+	return value
 }
 
 // getString attempts to resolve the named string toggle against the specified Entity.
 func (t *Toggles) getString(name string, entity *Entity) string {
 	if t == nil || t.Strings == nil {
 		log.Printf("string toggles are nil, resolving to default empty string value for toggle: %q", name)
 	}
+	var value string
 	if toggle, ok := t.Strings[name]; ok {
-		return toggle(entity)
+		value = toggle(entity)
+	}
+	log.Printf("resolved value of toggle %q; entity: %s; value: %s", name, entity, value)
```

Comment:
same a above.

Reply:
since it implements Stringer go will automatically call `String()` :)

similar to java and `toString()`

Reply:
Ah cool! thanks for clarifying